### PR TITLE
User defined snapshot

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://progress.opensuse.org/issues/60779
- Verification run:
WIP
https://openqa.suse.de/tests/4014439 15sp2 HA
https://openqa.suse.de/tests/4017761 15sp1 user_defined_snapshot
https://openqa.suse.de/tests/4017759 15 user_defined_snapshot
https://openqa.suse.de/tests/4014813 12sp5 user_defined_snapshot
https://openqa.suse.de/tests/4017757 12sp4 user_defined_snapshot
https://openqa.suse.de/tests/4017756 12sp3 user_defined_snapshot
http://10.100.12.155/tests/14795#step/zypper_migration/14 15sp2 HPC online_*_full_zypp migration
http://10.100.12.155/tests/14794#step/yast2_migration/14 15sp2 HPC online_*_full_yast migration
http://10.100.12.155/tests/14819#step/snapper_rollback/11 15sp2 HPC offline_*_def_full migration
http://10.100.12.155/tests/14813#step/opensuse_welcome/1 opensuse staging kde
http://10.100.12.155/tests/14811#step/first_boot/1 opensuse staging upgrade_staging
http://10.100.12.155/tests/14806#step/first_boot/1 opensuse staging cryptlvm
http://10.100.12.155/tests/14818#step/first_boot/1 opensuse staging cryptlvm uefi
http://10.100.12.155/tests/14805#step/first_boot/1 opensuse staging textmode
http://10.100.12.155/tests/14802#step/first_boot/1 opensuse staging autoyast_mini
http://10.100.12.155/tests/14790#step/first_boot/1 opensuse 15.2 gnome-gdm

### CODE REVIEW

**Comment by foursixnine** [comment]
@dzedro https://github.com/foursixnine/os-autoinst-distri-opensuse/commits/user_defined_snapshot i did a quick rebase

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -15,7 +15,7 @@ use bootloader_setup qw(boot_grub_item boot_local_disk stop_grub_timeout tianoco
 use testapi qw(is_serial_terminal :DEFAULT);
 use strict;
 use warnings;
-use utils;
+use utils qw(clear_console handle_emergency save_svirt_pty show_tasks_in_blocked_state type_line_svirt type_string_very_slow unlock_if_encrypted workaround_type_encrypted_passphrase);
```
Comment (foursixnine) [inline lib/opensusebasetest.pm]: i'd keep utils full here

File: tests/x11/user_defined_snapshot.pm [hunk 2]
```diff
@@ -65,8 +65,8 @@ sub run {
     send_key "alt-l";
     wait_serial("$module_name-0", 200) || die "'yast2 $module_name' didn't finish";
     $self->{in_wait_boot} = 1;
-    power_action('reboot', keepconsole => 1, textmode => 1);
-    $self->wait_grub(bootloader_time => 150);
+    power_action('reboot');
+    $self->wait_grub(bootloader_time => 200);
```
Comment (foursixnine) [inline tests/x11/user_defined_snapshot.pm]: See 20d6eba94

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (dzedro) [reply lib/opensusebasetest.pm]: sure

File: tests/x11/user_defined_snapshot.pm (hunk 2 above)
Comment (dzedro) [reply tests/x11/user_defined_snapshot.pm]: thx, updated

**Pull request closed without merging**